### PR TITLE
Update compile based commands

### PR DESCRIPTION
**Pull request by TinyPandas**
[U] - wandbox v1.0.3
[U] - promise implementation of fromString
[I] - compilers command for listing compilers

Closes: #30, #31, #39 and #40

### CODE REVIEW

File: src/commands/compile.ts [hunk 1]
```diff
@@ -41,76 +19,25 @@ const command: Command = {
...
-			}
-		}
+		fromString({ compiler: compiler, code: src.length == 0 ? "" : src === "0" ? "" : src, save: false })
+			.then(result => {
+				const embed = new MessageEmbed();
+
+				if (result.compiler_error) {
+					embed.setColor("#D95B18");
+					embed.setDescription("Compilation failed: compiler errors present.");
+					embed.addField("Compiler Error", result.compiler_error, false);
```
Comment (cntkillme) [inline src/commands/compile.ts:29]: As is the case with above, a compiler error can be part of the program error string for whatever reason for some languages.

```suggestion
					embed.addField("Compiler Error", result.compiler_error ?? result.program_error, false);
```

File: src/commands/compile.ts [hunk 2]
```diff
@@ -41,76 +19,25 @@ const command: Command = {
...
-				console.error(e);
-			}
-		}
+		fromString({ compiler: compiler, code: src.length == 0 ? "" : src === "0" ? "" : src, save: false })
+			.then(result => {
+				const embed = new MessageEmbed();
+
+				if (result.compiler_error) {
+					embed.setColor("#D95B18");
+					embed.setDescription("Compilation failed: compiler errors present.");
```
Comment (cntkillme) [inline src/commands/compile.ts:28]: The above change will combine compiler and program errors.

```suggestion
					embed.setDescription("Compilation failed: errors present.");
```

File: src/commands/compilers.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+				const listEmbed = new MessageEmbed({
+					timestamp: message.createdTimestamp,
+					color: "#FF000A",
+				});
+
+				list.forEach(compiler => {
+					listEmbed.addField(`${language} ${compiler.version}`, `Compiler: ${compiler.name}`, true);
+				});
+
+				message.reply(listEmbed).then(console.log.bind(console)).catch(console.error.bind(console));
```
Comment (cntkillme) [inline src/commands/compilers.ts:29]: Don't print anything to the console on success.

```suggestion
				message.reply(listEmbed).catch(console.error.bind(console));
```

File: src/commands/compile.ts [hunk 4]
```diff
@@ -41,76 +19,25 @@ const command: Command = {
...
+				} else {
+					embed.setColor("#24BF2F");
+					embed.setDescription("Compilation finished.");
+					embed.addField("Program Message", result.program_message, false);
+				}
+
+				message.reply(embed).then(console.log.bind(console)).catch(console.error.bind(console));
+			})
+			.catch(err => {
+				void message.reply(err);
```
Comment (cntkillme) [inline src/commands/compile.ts:39]: Avoid this as it relies on deprecated NodeJS functionality. In the future, unhandled promises might straight up terminate node.
```suggestion
				message.reply(err).catch(console.error.bind(console));
```

File: src/commands/compilers.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import { MessageEmbed } from "discord.js";
+import { getCompilers } from "wandbox-api-updated";
+import type { Command } from "../command";
+import { defaultArgumentParser } from "../parsers";
+
+const command: Command = {
+	name: "compilers",
+	description: "testing",
```
[the remark this answers is not in the document]
Comment (TinyPandas) [reply src/commands/compilers.ts:8]: 😅 So about this one, I made it in my private branch, thinking you were gonna beat me to it. I wound up copying and forgot about it.

Will fix.

**Pull request merged**